MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication.swift

```diff
@@ -56,6 +56,10 @@ public final class MSALNativeAuthPublicClientApplication: MSALPublicClientApplic
         self.controllerFactory = MSALNativeAuthControllerFactory(config: nativeConfiguration)
         self.inputValidator = MSALNativeAuthInputValidator()
 
+        if config.redirectUri == nil {
+            print(MSALNativeAuthErrorMessage.redirectUriNotSetWarning)
```

Comment:
@rodhan-ms , continuing here the conversation started by @diegojerezba , could you please get advice from Sergei for this specific case? And what MSAL Android did it use for the same log?
MSAL team never used NSLog, they have this rule: "You should only ever log to NSLog in debug mode to prevent leaking potentially sensitive information". But I'm not sure if we need to threat this case as a special case.

Reply:
Hi @antrix1989 can we get your input on this please? (See also https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1970#issuecomment-1889020399).

In short, basic Native Auth flows don't require the redirectUri to be set, especially when developers are taking their first steps with the SDK. However, as previously discussed, we want to make it clear to developers that they will need to add a redirect URI and implement the web fallback cases before their apps are in production. We agreed that we would log a warning to remind developers about this. The question now is where to put that log - should we use the regular MSAL log mechanism or should we make sure it shows in the Xcode console by using `print` directly.

Reply:
I suggest using the MSAL logger for all log messages. NSLog can significantly slow down performance.

> they will need to add a redirect URI and implement the web fallback cases before their apps are in production

If they have to do it eventually anyway, does it really matter whether it's done at the beginning or the end of their project? Should we simply add an assertion() that will fail in debug mode and indicate that a redirect URI is required?

Reply:
Thanks @antrix1989, I'll update the code to use the MSAL logger as suggested. 
The reason for logging and not, e.g. asserting is that we want to make early experimentation with the SDK as quick and easy as possible for developers. Thanks for your input.

Reply:
Thanks @antrix1989 , we wrote a document to describe the native auth approach at the web fallback (and redirect URI). https://github.com/microsoft/entra-previews/blob/PP4/docs/Native-Auth/Developer-guides/1-iOS-Swift/3-Additional-scenarios/7-support-web-fallback.md
We don't force the user to implement a fallback mechanism, but we inform of the issue that they can have if they don't do it.